### PR TITLE
Implements std.testing.expectEqual for tagged unions.

### PR DESCRIPTION
**Pull request by MasterQ32**

### CODE REVIEW

**Comment by andrewrk** [comment]
Nice work, can you use it in a test?

**Comment by MasterQ32** [comment]
I actually implemented it in a test of mine. I should add some tests on the test functions in the PR though...

my "real life test":
```zig

const Value = union(Type) {
    integer: i32,
    number: f32,
    … // tens of other members here, some with the same type
};

std.testing.expectEqual(obj.getProperty(10), @as(?Value, Value{ .integer = 15 }));
```

File: lib/std/testing.zig [hunk 1]
```diff
@@ -89,7 +89,26 @@ pub fn expectEqual(expected: var, actual: @typeOf(expected)) void {
             if (union_info.tag_type == null) {
                 @compileError("Unable to compare untagged union values");
             }
-            @compileError("TODO implement testing.expectEqual for tagged unions");
+
+            const TagType = @TagType(@typeOf(actual));
```
Comment (daurnimator) [inline lib/std/testing.zig]: Use `expected` here? ==> a coercion from actual -> expected makes more sense than expected -> actual.

Comment (MasterQ32) [reply lib/std/testing.zig]: Changed

**Comment by MasterQ32** [comment]
Added a test function, adjusted to daurnimators comment

**Pull request merged**